pkg/controller/installation/products/nexus/reconciler.go

```diff
@@ -0,0 +1,224 @@
+package nexus
+
+import (
+	"context"
+	"fmt"
+
+	nexus "github.com/integr8ly/integreatly-operator/pkg/apis/gpte/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/resources"
+
+	"github.com/integr8ly/integreatly-operator/pkg/controller/installation/marketplace"
+	"github.com/integr8ly/integreatly-operator/pkg/controller/installation/products/config"
+	coreosv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	pkgclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultInstallationNamespace = "nexus"
+	defaultSubscriptionName      = "nexus"
+	resourceName                 = "nexus"
+)
+
+type Reconciler struct {
+	coreClient    kubernetes.Interface
+	Config        *config.Nexus
+	ConfigManager config.ConfigReadWriter
+	mpm           marketplace.MarketplaceInterface
+	logger        *logrus.Entry
+}
+
+func NewReconciler(coreClient kubernetes.Interface, configManager config.ConfigReadWriter, instance *v1alpha1.Installation, mpm marketplace.MarketplaceInterface) (*Reconciler, error) {
+	config, err := configManager.ReadNexus()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not read nexus config")
+	}
+
+	if config.GetNamespace() == "" {
+		config.SetNamespace(instance.Spec.NamespacePrefix + defaultInstallationNamespace)
+	}
+	if err = config.Validate(); err != nil {
+		return nil, errors.Wrap(err, "nexus config is not valid")
+	}
+
+	logger := logrus.WithFields(logrus.Fields{"product": config.GetProductName()})
+
+	return &Reconciler{
+		coreClient:    coreClient,
+		ConfigManager: configManager,
+		Config:        config,
+		mpm:           mpm,
+		logger:        logger,
+	}, nil
+}
+
+func (r *Reconciler) Reconcile(inst *v1alpha1.Installation, serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	ctx := context.TODO()
```

Comment:
This should be using a passed in context now. Can you check you're rebased against master?